### PR TITLE
Added support for dynamically setting tfplan and tfvars file names wh…

### PR DESCRIPTION
**Pull request by DeclanKainos**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-9663

### Change description ###
Added support for dynamically setting terraform Plan out and terraform var file names, depending on whether multi-regions are specified; this allows us to retain symplified tf-cmt tagging and also allocate one tf-vars file in cases where multi-regions are used.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: steps/terraform.yaml [hunk 1]
```diff
@@ -109,12 +120,12 @@ steps:
...
       commandOptions: >
-        -out tfplan-${{ parameters.environment }}-${{ lower(replace(parameters.location, ' ', '')) }}-${{ parameters.component }}
+        -out tfplan-$(tfPlanName)
         -var env=${{ parameters.environment }}
         -var builtFrom=$(Build.Repository.Name)
         -var product=${{ parameters.product }} ${{ parameters.planCommandOptions }}
-        -var-file $(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/environments/${{ parameters.environment }}/${{ parameters.environment }}.tfvars
-      publishPlanResults: "${{ parameters.environment }}-${{ lower(replace(parameters.location, ' ', '')) }}-${{ parameters.component }}"
+        -var-file $(System.DefaultWorkingDirectory)/$(buildRepoSuffix)/environments/${{ parameters.environment }}/$(tfVarsName).tfvars
+      publishPlanResults: "$(TtfPlanName)"
```
[the remark this answers is not in the document]
Comment (DeclanKainos) [reply steps/terraform.yaml]: Corrected

**Comment by madjava** [comment]
**NOTE:** README would be updated once we get the approves we need before merge as this would become the pattern when a repo needs that separation between ukw and uks for its envs files

**Review by timja — COMMENTED** [review]
Where has this been tested? multi region added and not?

File: README.md [hunk 2]
```diff
@@ -93,4 +93,19 @@ Template requires the below folder structure for the build repository.
     │   │    └── *.tfvars
     ├── azure_pipeline.yaml
     ├── .terraform-version                                 # terraform version (read by tfenv)
+
+### Multi Regions:
```
Comment (timja) [inline README.md:97]:
```suggestion
### Multiple regions
```

File: README.md [hunk 3]
```diff
@@ -93,4 +93,19 @@ Template requires the below folder structure for the build repository.
     │   │    └── *.tfvars
     ├── azure_pipeline.yaml
     ├── .terraform-version                                 # terraform version (read by tfenv)
+
+### Multi Regions:  
+
+Some repos have separate `tfvars` files based on region i.e. `UK South` or `UK West` and having all enviroment
+variables in a single `tfvars` file may not meet the requirement.
+If you need to use a different set of variables for your build then you can pass the `multiRegion` parameter and
+set his as `true`, the default is `false`.
```
Comment (timja) [inline README.md:102]:
```suggestion
set this as `true`, the default is `false`.
```

File: README.md [hunk 4]
```diff
@@ -93,4 +93,19 @@ Template requires the below folder structure for the build repository.
     │   │    └── *.tfvars
     ├── azure_pipeline.yaml
     ├── .terraform-version                                 # terraform version (read by tfenv)
+
+### Multi Regions:  
+
+Some repos have separate `tfvars` files based on region i.e. `UK South` or `UK West` and having all enviroment
```
Comment (timja) [inline README.md:99]:
```suggestion
Some repositories have separate `tfvars` files based on region i.e. `UK South` or `UK West` and having all enviroment
```

File: README.md [hunk 5]
```diff
@@ -93,4 +93,19 @@ Template requires the below folder structure for the build repository.
...
+variables in a single `tfvars` file may not meet the requirement.
+If you need to use a different set of variables for your build then you can pass the `multiRegion` parameter and
+set his as `true`, the default is `false`.
+
+Using multi region requires the below environments folder structure for the build repository.
```
Comment (timja) [inline README.md:104]:
```suggestion
Using multiple region support requires the below environments folder structure for the build repository.
```

**Comment by madjava** [comment]
> Where has this been tested? multi region added and not?

You can find an example in this [panorama repo ](DTSPO-9663-template-file-cleanup)PR we hope will begin to use this addition once alls in master

**Review by timja — APPROVED** [review]

**Review by madjava — APPROVED** [review]

**Pull request merged**